File: tools/project-memory-context/tests/fixtures/sample.kt
interface Greetable {
    fun greet(): String
}

data class Animal(val name: String) : Greetable {
    override fun greet() = "Hello, I'm $name"
}

object Singleton {
    fun getInstance() = this
}

fun standaloneFunction(x: Int, y: Int): Int = x + y

private fun privateHelper(): Unit {}

sealed class Result<T>

enum class Status { ACTIVE, INACTIVE }

File: tools/project-memory-context/tests/fixtures/sample.swift
protocol Greetable {
    func greet() -> String
}

public class Animal: Greetable {
    public func greet() -> String { return "hello" }
    func privateMethod() {}
}

public struct Point {
    var x: Double
    var y: Double
    public func distance() -> Double { return 0.0 }
}

extension Animal {
    func extended() {}
}

public func standaloneFunction(x: Int, y: Int) -> Int {
    return x + y
}

enum Status {
    case active
    case inactive
}
